UefiPayloadPkg/GraphicsOutputDxe: Allow for framebuffer at an offset from BAR

mkopec opened the pull request:
On Meteor Lake the framebuffer is at BAR2 + 0x800000. Allow for this case when validating the framebuffer info received from coreboot.

Fixes graphics output on Meteor Lake platforms.

mkopec committed:
UefiPayloadPkg/GraphicsOutputDxe: Allow for framebuffer at an offset from BAR

On Meteor Lake the framebuffer is at BAR2 + 0x800000. Allow for this
case when validating the framebuffer info received from coreboot.

Fixes graphics output on Meteor Lake platforms.

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

paulmenzel commented:
s/graphcics/graphics/

Maybe mention the addresses, so it is visible where it fails right now?

mkopec commented:
> s/graphcics/graphics/

Fixed, thanks

> Maybe mention the addresses, so it is visible where it fails right now?

Added the offset to the commit and PR messages. The offset is defined here in coreboot: https://github.com/coreboot/coreboot/blob/0c1897e4fd40b72a7096848a5b65151a9d85ab97/src/soc/intel/meteorlake/Kconfig#L387-L389

On NovaCustom V540TU the actual framebuffer address comes out to `0x80800000`
